Issue #691 Clean up doc set

thatdocslady opened the pull request:
Fixes issue #691 

This PR intends to group together style and formatting conventions based on recent work on #618.

Currently the PR contains only the topics in the Getting Started section, as well as the topic map.

Question for @hferentschik @LalatenduMohanty: it seems to take about a day for fixing up a category (i.e. Getting Started) and testing everything. Should I create multiple commits, each with its own category fixes? I am trying to avoid merge/push conflicts if other people are working on the same topics.

hferentschik commented:
> Should I create multiple commits, each with its own category fixes?

Sure multiple commits is fine. One commit per category is fine. We can squash in the end (if we want).

hferentschik commented on docs/source/getting-started/index.adoc:
```diff
@@ -5,8 +5,8 @@
 This section contains information about installing Minishift, quickstart steps, and sample
```
What's about we take care of "line per sentence" as well. A fair bit of the docs are already written this way. It would be nice to align the rest as well.

thatdocslady replied:
Sure, I just realized now that we never settled on this, but the openshift folks said it doesn't matter to publishing so I can fix it here. We should also add this to our own doc guidelines probably, no?

thatdocslady commented:
Ok I powered through and managed to fix up everything (tomorrow I have a late start and I want to try and get a review going ASAP). Removing the WIP because it's now ready for review.

A few notes to considering during the review:

- There are 4 commits, one for each category/directory. The only exception is that the Getting Started commit also has the topic map, I couldn't figure out how to extract it after committing it, and I have a feeling it's not a huge deal?
- There's no command reference topic for `minishift oc-env`, but there are some references to it in the docs. Not sure what happened to it or whether we replaced it with something else?
- Legally we are supposed to use "container registry" and not "docker registry" when talking about downstream offerings, but I'm not sure this is relevant for upstream/community editions. Thoughts?
- I didn't touch the CI topic in Contributing because there's an open PR for issue #426, will fix this later after that PR is merged.
- While I was fixing the style and formatting, I also tweaked a few terms that needed standardizing. Please let me know if you have questions about any of these (i.e. changed "plugin" to "plug-in").

hferentschik commented:
> There are 4 commits, one for each category/directory. The only exception is that the Getting Started commit also has the topic map, I couldn't figure out how to extract it after committing it, and I have a feeling it's not a huge deal?

It is not. In fact I think when we apply this we will squash anyways. 

> There's no command reference topic for minishift oc-env, but there are some references to it in the docs. Not sure what happened to it or whether we replaced it with something else?

I would think you have not re-generated the command ref files since _oc-env_ got added. 

    $ make clean clean_docs 
    $ make gen_docs

After that you should see them as well.

> Legally we are supposed to use "container registry" and not "docker registry" when talking about downstream offerings, but I'm not sure this is relevant for upstream/community editions. Thoughts?

I am fine to stick with Docker registry for now.

> I didn't touch the CI topic in Contributing because there's an open PR for issue #426, will fix this later after that PR is merged.

+1 In fact I would assume that won't be necessary. @Preeticp is re-writing the whole thing. I would think it will in the end adhere to the standards as well.

> While I was fixing the style and formatting, I also tweaked a few terms that needed standardizing. Please let me know if you have questions about any of these (i.e. changed "plugin" to "plug-in").

Sounds good.

hferentschik commented on docs/source/contributing/developing.adoc:
```diff
@@ -111,18 +111,18 @@ $ glide install -v
  Glide will recognize that there is no lock file and recalculate the required dependencies.
 
-. Check-in the updated *_glide.yaml_* and *_glide.lock_*.
+. Check-in the updated *_glide.yaml_* and *_glide.lock_* files.
```
_files_ is imo superfluous here. It just makes it sounds odd, even though it is correct. I am saying "I check-in start.go" not "I am checking in the start.go file". That's implicit.

thatdocslady replied:
You're right that when you refer to the proper noun you don't need a qualifier, but in this case you used the definite article which does require it, and you're using an adjective ("updated") which requires the definite article. These are cascading grammatical dependencies. 

And a general comment, even though you personally use certain language to discuss things, try to remember that not everyone does, and we need to be prepared for different types of readers, which means keeping the language and grammar as explicit as possible.

hferentschik replied:
> And a general comment, even though you personally use certain language to discuss things, try to remember that not everyone does, and we need to be prepared for different types of readers

Agree. On the same note, please remember that we are targeting developers who have partly their own lingo. What might make sense for a non technical user or non-dev, will sound awkward for a developer.

hferentschik commented on docs/source/contributing/developing.adoc:
```diff
@@ -194,11 +193,11 @@ $go test -v <relative path of package> -run "Test<Regex pattern to match tests>"
-The test cases are defined in *_test/integration/+*+.feature_*.
-Per default, the tests are executed against the binary created by `make build`, that is *_$GOPATH/bin/minishift_*.
+Integration tests utilize link:https://github.com/DATA-DOG/godog[Godog], which uses Gherkin (Cucumber) to define test cases.
+The test cases are defined in *_test/integration/features_*.
```
That's not correct anymore. Why did you change this? There is no directory _test/integration/features_, there is a __test/integration_ and it contains several files with the feature suffix. This is what was expressed in the existing expression. Also, it needs the extra '+' since one needs to escape the '*'. Let's change this back please.

thatdocslady replied:
Actually when I checked why the weird string, I saw that there is this directory in my local minishift repo, which is why I changed it. Attaching screenshot, notice the breadcrumb. I can add back the wildcard annotation but the directory structure looks ok to me.

thatdocslady replied:
![screenshot from 2017-06-22 15-38-35](https://user-images.githubusercontent.com/6662054/27437666-45e1f0da-5763-11e7-80db-8c25f3288f49.png)

hferentschik replied:
I take this back. The directory does exist. I believe this got refactored at some stage. The idea with *.feature was to make explicit what the test files are, but I guess we don't need it.

hferentschik commented on docs/source/getting-started/quickstart.adoc:
```diff
@@ -50,17 +45,15 @@ Starting local OpenShift cluster using 'kvm' hypervisor...
 ====
 
-. Use link:../command-ref/minishift_oc-env{outfilesuffix}[`minishift oc-env`] to display the command you need to type into your shell in order to add the _oc_ binary to your PATH.
+. Use xref:../command-ref/minishift_oc-env.adoc#minishift_oc-env[`minishift oc-env`] to display the command you need to type into your shell in order to add the *oc* binary to your `PATH` environment variable.
```
should be \`oc\` instead of \*oc\*. It's a binary.

thatdocslady replied:
I'm replying once here but it should cover all of the instances where you commented about this. In the openshift doc guidelines, the instruction is to tag "System term, GUI menu items and buttons, daemon, service, or software package" with **, while "Inline commands, operations, literal values, environment variables, parameters, settings, flags, and user input" should use ``. AFAIK a binary is a software package so that's why it gets the ** tag. When referring to env vars or commands I use ``. 

Reference: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference

hferentschik replied:
fair enough

hferentschik commented on docs/source/getting-started/setting-up-driver-plugin.adoc:
```diff
@@ -1,25 +1,23 @@
 toc::[]
 
-[[hypervisor-drivers-overview]]
+[[setup-driver-plugin-overview]]
```
good catch. There is no need so to make the overview anchor unique. 'overview' will just work fine.

thatdocslady replied:
Thanks, and actually even overview anchor IDs need to be unique across the entire repository (as in no matter the hierarchy). This is according to the openshift guidelines: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#unique-ids

hferentschik replied:
Well, it is not required and I think the section you are referring to is misleading. Uniqueness is given by the full context. There would be no loss in information here, nor would there be an issue with other tooling. But let's move on

hferentschik commented on docs/source/getting-started/setting-up-driver-plugin.adoc:
```diff
@@ -108,25 +105,23 @@ $ sudo chown root:wheel $(brew --prefix)/opt/docker-machine-driver-xhyve/bin/doc
+The following steps explain the installation of the *docker-machine-driver-xhyve* binary to the *_/usr/local/bin/_* directory:
 
-. Download the `docker-machine-driver-xhyve` binary using:
+. Download the *docker-machine-driver-xhyve* binary using:
```
back-ticks, we are talking about a binary here.

hferentschik commented on docs/source/getting-started/setting-up-driver-plugin.adoc:
```diff
@@ -108,25 +105,23 @@ $ sudo chown root:wheel $(brew --prefix)/opt/docker-machine-driver-xhyve/bin/doc
 ----
 
-. Enable root access for the `docker-machine-driver-xhyve` binary and add it to the default wheel group:
+. Enable root access for the *docker-machine-driver-xhyve* binary and add it to the default wheel group:
```
Same as above. Why are you changing the formatting here?

hferentschik commented on docs/source/getting-started/setting-up-driver-plugin.adoc:
```diff
@@ -60,13 +57,13 @@ $ newgrp libvirtd
 === On Fedora
 
-.  Install `libvirt` and `qemu-kvm` on your system:
+.  Install *libvirt* and *qemu-kvm* on your system:
```
hmm, not sure whether one should consider libvirt and qemu-kvm as actual binaries.

hferentschik commented on docs/source/getting-started/setting-up-driver-plugin.adoc:
```diff
@@ -140,9 +135,9 @@ $ sudo chmod u+s /usr/local/bin/docker-machine-driver-xhyve
-The downloaded `docker-machine-driver-xhyve` binaries are compiled against a specific
-version of OS X. It is possible that the driver fails to work after an OS X version
-upgrade. In this case you can try to compile the driver from source:
+The downloaded *docker-machine-driver-xhyve* binaries are compiled against a specific version of OS X.
```
Ok, let's align also how we refer to OS X. According to the latest, it should be macOS. I believe we might use it once or twice, but as we see here, not across the board. Let's do a find&replace of OS X and replace it with macOS.

thatdocslady replied:
Ugh, I am not impressed with this capitalization, but I don't work at Apple so not much I can do to protest ;) will search and replace.

hferentschik commented on docs/source/openshift/openshift-client-binary.adoc:
```diff
@@ -12,12 +12,13 @@ toc::[]
-It is located in the  _~/.minishift/cache/oc/{openshift-version}_ directory, provided you use Minishift's default version of OpenShift.
-You can add this binary to your `PATH` using xref:../command-ref/minishift_oc-env.adoc#minishift-oc-env[`minishift oc-env`] which displays the command you need to type into your shell.
-The output of `oc-env` differs depending on the operating system and the shell type.
+The *oc* binary is located in the  *_~/.minishift/cache/oc/{openshift-version}_* directory, assuming that you use Minishift's default version of OpenShift.
```
backticks for _oc_

hferentschik commented on docs/source/openshift/openshift-client-binary.adoc:
```diff
@@ -27,26 +28,26 @@ export PATH="/Users/john/.minishift/cache/oc/v1.5.0:$PATH"
-As part of the `minishift start` command there is also a `minishift` https://docs.openshift.org/latest/cli_reference/manage_cli_profiles.html[oc context] created.
-This context contains the configuration to communicate with your OpenShift cluster.
+As a part of the `minishift start` command, a `minishift` context is also created.
+This context contains the configuration to communicate with your OpenShift cluster, and is similar to an OpenShift https://docs.openshift.org/latest/cli_reference/manage_cli_profiles.html[CLI profile].
```
"and is similar"? It is not similar to a CLI profile, it **is** a CLI profile.

hferentschik commented on docs/source/openshift/openshift-client-binary.adoc:
```diff
@@ -27,26 +28,26 @@ export PATH="/Users/john/.minishift/cache/oc/v1.5.0:$PATH"
 
 [[minishift-oc-context]]
-== Minishift oc context
+== Minishift oc Context
```
Not really happy about the capitalisation here.

thatdocslady replied:
We're supposed to use title caps in all headings, according to the openshift guidelines https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#topic-titles-and-section-headings

In fact, I'd like to suggest that we ditch the term "oc context", now that I dug a little into it and you also confirmed that it's a CLI Profile, and I would propose that we rename it as such so that we're aligned with the terms. I don't know where the term "oc context" came from" but it's not accurate, even though the command uses that term (which IMO should have been aligned in the first place but I guess it's too late now to change that).

hferentschik replied:
> We're supposed to use title caps in all headings, according to the openshift guidelines

I know, however, in this context it is a bit unfortunate that 'oc' is lowercase. Never mind.

> In fact, I'd like to suggest that we ditch the term "oc context", now that I dug a little into it and you also confirmed that it's a CLI Profile, and I would propose that we rename it as such so that we're aligned with the terms.

I am happy to rename the section "Minishift CLI Profile", however, we also need to keep the term 'context'. The term is correct as well and probably closer to what actually happens on a technical level - https://docs.openshift.org/latest/cli_reference/manage_cli_profiles.html

So +1 for changing the title, -1 for removing the mention to 'context'.

hferentschik commented on docs/source/openshift/openshift-client-binary.adoc:
```diff
@@ -109,36 +108,29 @@ $ minishift logs
 
 [[example-config-cors]]
-=== Example: Configuring cross-origin resource sharing
+.Example: Configuring cross-origin resource sharing
```
This makes the entry disappear from the TOC and makes this example hard to find. I think it is better to keep these as sections. It gives us entries in the TOC and I also feel it is a bit too much content for the '.Example' form.

thatdocslady replied:
Good point, I'll convert these back to headings and rephrase the lead-in sentence a bit to work with it.

hferentschik replied:
thanks

hferentschik commented on docs/source/openshift/openshift-client-binary.adoc:
```diff
@@ -148,42 +140,37 @@ $  minishift openshift config set --patch '{"corsAllowedOrigins": [".*"]}'
 
 [[example-change-openshift-routing-suffix]]
-=== Example: Changing the OpenShift routing suffix
+.Example: Changing the OpenShift routing suffix
```
Same as above.

hferentschik commented on docs/source/openshift/openshift-docker-registry.adoc:
```diff
@@ -11,37 +11,42 @@ toc::[]
-. Make sure your shell is configured to  link:../using/docker-daemon{outfilesuffix}[reuse the Minishift docker daemon].
-. Login to the docker registry
-
+. Start Minishift and add the *oc* binary to the PATH. For a detailed example, see the xref:../getting-started/quickstart.adoc#quickstart[Minishift Quickstart] section.
```
formatting of 'oc'

hferentschik commented on docs/source/using/addons.adoc:
```diff
@@ -31,65 +31,60 @@ oc adm policy add-scc-to-group anyuid system:authenticated
 oc::
-If the add-on command starts with _oc_, it uses the _oc_ binary that is cached on your host to execute
-the specified `oc` command. This is similar to running `oc --as system:admin ...` from the command-line.
+If the add-on command starts with `oc`, it uses the *oc* binary that is cached on your host to execute the specified `oc` command.
```
I think it formatting should be the other way around. First use of oc \*oc\* (here we are referring to the "text" oc), second use \`oc\` since we are referring to the binary/command

hferentschik commented on docs/source/using/addons.adoc:
```diff
@@ -31,65 +31,60 @@ oc adm policy add-scc-to-group anyuid system:authenticated
 echo::
-If the add-on command starts with _echo_, the arguments following the _echo_ command are printed
-to the console. This can be used to provide additional feedback during add-on execution.
+If the add-on command starts with `echo`, the arguments following the `echo` command are printed to the console.
```
\*echo\*, we are not referring to binary or command here.

hferentschik commented on docs/source/using/addons.adoc:
```diff
@@ -31,65 +31,60 @@ oc adm policy add-scc-to-group anyuid system:authenticated
 
 [[example-addon-routing-variable]]
 .Example: Usage of the routing suffix variable
+
```
I am not a big fan of pulling this into the example. The example is the console listing, not the explaining text which imo should live outside. This applies to the other examples as well.

thatdocslady replied:
You're right, I missed that one. Will pull the text out and rephrase to flow into the example heading.

hferentschik commented on docs/source/using/addons.adoc:
```diff
@@ -100,53 +95,52 @@ $ openshift admin ca create-server-cert \
-
 [[table-supported-addon-variables]]
 .Supported add-on variables
+
```
-1, I don't think this is a good form.

thatdocslady replied:
What do you mean "not good form"? I didn't change the heading of the table or the structure, I only removed the unnecessary definite articles, the text is clear without them and there's no need for "The" in the beginning of every line.

hferentschik replied:
The comment is the same as for the example ones. You are pulling text into the table, similar as to the examples above.

hferentschik commented on docs/source/using/addons.adoc:
```diff
@@ -100,53 +95,52 @@ $ openshift admin ca create-server-cert \
+This command prints a list of installed add-ons. You should at least see the *anyuid* add-on listed.
 This is an important add-on that allows you to run images that do not use a pre-allocated UID.
-By default, this is not allowed in OpensShift.
+By default, this is not allowed in OpenShift.
```
:-)

hferentschik commented on docs/source/using/addons.adoc:
```diff
@@ -181,48 +176,46 @@ By default, an add-on has the priority 0. Add-ons with a lower priority value ar
 [[example-priority-list]]
 .Example: List command output with explicit priorities
 
+In this example, the *anyuid*, *registry*, and *eap* add-ons are enabled with the respective priorities of 0, 5 and 10.
```
Same as with other tables and examples. This text belongs outside of the example/table scope. I double checked with the rest of docs.openshift.org and there it is also outside.

hferentschik commented on docs/source/using/host-folders.adoc:
```diff
@@ -12,60 +12,71 @@ toc::[]
 Host folders are directories on the host which are shared between the host and the Minishift VM.
 They allow for a two way file synchronization between host and VM.
-The following sections discuss the various types of host folders, driver provided host folders, as well as the Minishift `hostfolder` command.
+The following sections discuss the various types of host folders, driver provided host folders, as well as the `minishift hostfolder` command.
```
I guess both works.

thatdocslady replied:
I made similar changes in lots of places where people wrote only the sub-command without the parent, and I did this because of the likelihood that someone will copy and paste the command text, and if we only write something like "get" instead of "minishift config get" then the command wouldn't work.

hferentschik commented on docs/source/using/host-folders.adoc:
```diff
@@ -12,60 +12,71 @@ toc::[]
 It is also possible to use CIFS on OS X and Linux.
-On the former you can enable CIFS based shares under _System Preferences > Sharing_.
-Refer to link:https://support.apple.com/en-us/HT204445[How to connect with File Sharing on your Mac] for detailed setup instructions.
+On OS X, you can enable CIFS-based shares under *System Preferences > Sharing*.
```
Here is another OS X

hferentschik commented on docs/source/using/host-folders.adoc:
```diff
@@ -186,26 +199,23 @@ No host folders defined
 
-Mac OS X also has a built-in
-SSH server. To use it, make sure that _Remote Login_ is enabled in _System Preferences > Sharing_.
+Mac OS X also has a built-in SSH server.
```
OS X

hferentschik commented:
@thatdocslady very nice indeed. Things are definitely taking shape.

thatdocslady commented:
@hferentschik I reviewed your feedback and replied inline wherever possible. The only thing about the oc-env comment, is that I actually I regenerated the docs many times (because I had to test every single link and make sure everything still compiles after changing the formatting), it's minishift itself (i.e. make clean) that I didn't run lately, I didn't think that I needed to clean up that deep. Now it works :)

hferentschik commented:
>  I reviewed your feedback and replied inline wherever possible. Let me know if you want to bjn about any of these, I hope my replies are clear and helpful.

I am fine. I get your formatting of 'oc' now. There are still several other things which needs updating, in particulate the examples and tables. Once you pushed an update I can have another look and if there are still question we can have a chat. WDYT?

thatdocslady commented:
@hferentschik implemented your feedback, and specifically:

- Aligned macOS term
- Changed oc context section title and rephrased to lead into "context" terminology
- Fixed examples and tables

I think this and a few other fixes basically cover what you asked for, I'm happy with how things are looking so if it looks good to you I think we can merge it. Of course rebased etc. :)

hferentschik commented:
> I think this and a few other fixes basically cover what you asked for

+1, thanks for applying the fixes

> I'm happy with how things are looking 

Same here

> So if it looks good to you I think we can merge it.

Will rebase, squash and merge.

hferentschik commented:
Rebased, squashed and merged. Thanks.

LalatenduMohanty commented on docs/source/contributing/developing.adoc:
```diff
@@ -131,10 +131,9 @@ If you seeing Glide errors during `make vendor`, you can clear the Glide cache v
 
 [[build-minishift-binary]]
-=== Building the Minishift binary
+=== Building the Minishift Binary
```
@thatdocslady I confused why `the` has small `t` but other words have capital letter as starting letter.

gbraad replied:
Articles are usually written as lowercase; 'a', 'an', and 'the'.

The point here would be, consistency. In the documentation there is
  * Building the Minishift binary
  * Running the Minishift binary

thatdocslady replied:
Right @LalatenduMohanty , the rules for title-case capitalization are not to capitalize articles and also some prepositions such as "in", "on". 

@gbraad I don't know what you are referring to, I see the titles on the master branch and they're capitalized correctly. Where did you spot this?

gbraad replied:
@thatdocslady I mention that they are consistent. as the report would have only pick one of the both cases.
